packages/backend/src/peripherals/arbitrum/multicall/ArbitrumMulticall.ts

```diff
@@ -0,0 +1,100 @@
+import { ChainId, EthereumAddress } from '@l2beat/shared-pure'
+
+import { EthereumClient } from '../../ethereum/EthereumClient'
+import {
+  MulticallEncoder,
+  MulticallRequest,
+  MulticallResponse,
+} from './interfaces'
+import { arbitrumMulticallEncoder } from './MulticallEncoder'
+
+/**
+ * @see https://arbiscan.io/address/0x842eC2c7D803033Edf55E478F461FC547Bc54EB2
+ */
+export const ARBITRUM_MULTICALL_BATCH_SIZE = 150
+export const ARBITRUM_MULTICALL_BLOCK = 821923
+export const ARBITRUM_MULTICALL_ADDRESS = EthereumAddress(
+  '0x842eC2c7D803033Edf55E478F461FC547Bc54EB2',
+)
+
+export class ArbitrumMulticallClient {
+  private readonly chainId = ChainId.ARBITRUM
+
+  static forMainnet(arbitrumClient: EthereumClient) {
```

Comment:
why do we need this static method?

Reply:
Since we will be using only mainnet variant for now, and we have no configuration management for the multicalls yet, I wanted to prevent importing those magic numbers across the code

Reply:
aah, I see, the class `EthereumClient` was misleading, maybe we should think about renaming it in the future to something like `EVMClient`